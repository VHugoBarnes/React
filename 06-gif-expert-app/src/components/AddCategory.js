import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
    
    const [inputValue, setInputValue] = useState('');

    // Cada que se introduce una letra se escribe en la caja de texto del input
    const handleInputChange = ( e ) => {
        // console.log(e.target.value);
        setInputValue(e.target.value);
    };

    const handleSubmit = ( e ) => {
        e.preventDefault(); // previene que se cargue toda la pÃ¡gina
        // console.log('Submit hecho');
        if(inputValue.trim().length >= 1){
            setCategories( cats => [inputValue, ...cats] );
            setInputValue('');
        }
    };
    
    return (
        <form onSubmit={ handleSubmit } >
            <input
                type="text"
                value={ inputValue }
                onChange={ handleInputChange }
                placeholder='Buscar...'
            />
        </form>
    )
};

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired,
};
