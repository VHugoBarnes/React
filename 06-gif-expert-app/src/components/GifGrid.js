import React, { useState, useEffect } from 'react'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {

    const [images, setImages] = useState([]);

    useEffect( () => {
        getGif();
    }, [] );

    const getGif = async() => {
        const url = 'https://api.giphy.com/v1/gifs/search?q=the+office&limit=6&api_key=69P2JB6qo3viyWDvHpnvtSyvEGFzNfzb';
        const resp = await fetch( url );
        const {data} = await resp.json();

        const gifs = data.map( img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images.downsized_medium.url
            }
        } );

        console.log(gifs);
        setImages( gifs );
    }

    return (
        <div>
            <h3>{ category }</h3>
                {
                    images.map( (i) => 
                    (<GifGridItem 
                        key={i.id}
                        {...i}
                    />
                    )
                    )
                }
        </div>
    )
}
