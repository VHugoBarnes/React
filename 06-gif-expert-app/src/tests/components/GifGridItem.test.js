/**
 * Debe mostrar el componente correctamente
 * shallow
 * wrapper
 * wrapper .toMatchSnapshot
 */

 import { shallow } from 'enzyme';
 import '@testing-library/jest-dom';
 import React from 'react';
 import { GifGridItem } from '../../components/GifGridItem';

 describe('Pruebas en el componente <GifGridItem />', () => {

     const title = 'Un titulo';
     const url = 'http://localhost/algo.jpg';
     let wrapper = shallow( <GifGridItem title={title} url={url}/> );

    beforeEach( () => { wrapper = shallow( <GifGridItem title={title} url={url}/> ) } );
    
    test('Debe mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
 }); 
 