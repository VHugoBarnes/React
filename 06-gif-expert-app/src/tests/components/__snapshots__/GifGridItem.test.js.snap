// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifGridItem /> Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeInLeft"
>
  <img
    alt="Un titulo"
    src="http://localhost/algo.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
